src/types/smartSearch.ts

```diff
@@ -46,8 +47,18 @@ export interface UserFilterConfig {
     is_user: boolean;
 }
 
+export interface CampaignParticipationConfig {
+    state: 'booked' | 'signed_up';
+    operator: 'in' | 'notin';
+    campaign?: number;
+    activity?: number;
+    location?: number;
+    after?: string;
+    before?: string;
+}
+
 export type AnyFilterConfig = DefaultFilterConfig | MostActiveFilterConfig |
-    RandomFilterConfig | UserFilterConfig// Add all filter objects here
+    RandomFilterConfig | UserFilterConfig | CampaignParticipationConfig // Add all filter objects here
```

Comment:
This list is getting long. We should probably order it alphabetically at some point.